src/app/WriteClient.h

```diff
@@ -214,11 +214,11 @@ class WriteClient : public Messaging::ExchangeDelegate
      *  handle calling Shutdown on itself once it decides it's done with waiting
      *  for a response (i.e. times out or gets a response). Client can specify
      *  the maximum time to wait for response (in milliseconds) via timeout parameter.
-     *  Default timeout value will be used otherwise.
+     *  If the timeout is missing or is set to System::Clock::kZero, a value based on the MRP timeouts of the session will be used.
      *  If SendWriteRequest is never called, or the call fails, the API
      *  consumer is responsible for calling Shutdown on the WriteClient.
      */
-    CHIP_ERROR SendWriteRequest(const SessionHandle & session, System::Clock::Timeout timeout = kImMessageTimeout);
+    CHIP_ERROR SendWriteRequest(const SessionHandle & session, System::Clock::Timeout timeout = System::Clock::kZero);
```

Comment:
Think of it as a pivot from "Response Timeout" to "Execution Timeout".

Reply:
Application-wise, I think it's only intuitive to think of total time-outs in most cases. Considering comms delays and execution delays separately is not trivial. I don't know the magical answer, but changing to just execution timeout will be really hard to interpret for a really sleepy device. Perhaps it should be a "total timeout" but with utilities to determine a session's comms-timeout so that callers can decide to account for it?

Reply:
In most situation, `ResponseTimeout` have totally different meaning against ack timeout, it is tightly bonded to user experience that reflect how long the UI raise an error after an user action. Take TCP as an example, application layer timer doesn't have any relationship with underlying TCP timer. On the other hand, I wonder how measuring execution time on the other end is helpful.